avocado/utils/data_structures.py

```diff
@@ -33,6 +34,24 @@ def ordered_list_unique(object_list):
     return [x for x in object_list if not (x in seen or seen_add(x))]
 
 
+def geometric_mean(values):
+    """
+    Evaluates the geometric mean for a list of numeric values.
+    This implementation is slow.
+    :param values: List with values.
+    :return: Single value representing the geometric mean for the list values.
+    :see: http://en.wikipedia.org/wiki/Geometric_mean
+    """
+    try:
+        values = [int(value) for value in values]
+    except ValueError:
+        raise ValueError('Invalid inputs. Provide valid inputs')
```

Comment:
Well personally I don't think this is necessary, one should see the problem from the line itself. But if you believe this is important, please include the `values` in the exception so one knows what caused the exception.